web/modules/custom/rdf_etl/src/ProcessStepBase.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\rdf_etl;
+
+use Drupal\Component\Plugin\PluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\rdf_etl\Plugin\EtlProcessStepInterface;
+
+/**
+ * Base class for the for process step plugins supporting forms.
+ */
+abstract class ProcessStepBase extends PluginBase implements EtlProcessStepInterface {
+  use StringTranslationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+
+    $this->configuration += $this->defaultConfiguration();
```

Comment:
This should be replaced by `$this->setConfiguration($configuration);` and the merge of default configuration should be mode there:

```php
public function setConfiguration(array $configuration): void {
  $this->configuration = $configuration + $this->defaultConfiguration();
  // Or even use NestedArray::mergeDeep()....
}
```